book.rb

```diff
@@ -0,0 +1,15 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    rental.book = self
+    @rentals.push(rental)
+  end

```

Comment:
- [ ] The `Rental` class is the go-between class between the `Person` and the `Book` classes. When creating the `Rental` class in the `initialize` method, you set the `Person` and `Book` instances to the arguments passed and also update the instances' arrays. That is the correct implementation 👍. This means you don't need to repeat the same logic in the `add_rental` method of the `Book` class since the `Rental` class has that handled. 

In the `add_rental` method, all you have to do is add the `date` parameters and the `person` parameters, then create a new instance of the `Rental` class because the logic in the `initialize` method of the `Rental` class will handle the implementation of the many-to-many relationship between the `Book` and `Person` classes. 👍 

#### Hint:

```ruby
def add_rental(date, person)
  Rental.new(date, self, person)
end
```